crates/ruff_python_semantic/src/scope.rs

```diff
@@ -19,6 +19,8 @@ pub struct Scope<'a> {
     /// A list of star imports in this scope. These represent _module_ imports (e.g., `sys` in
     /// `from sys import *`), rather than individual bindings (e.g., individual members in `sys`).
     star_imports: Vec<StarImportation<'a>>,
+    /// A list of all names declared in the scope.
+    declared_symbols: Vec<&'a str>,
```

Comment:
Would it make sense to instead track deleted `symbol`s, considering that deletions are rare (so that the vec is empty 99% of the cases)?

Reply:
Yes, I did that initially. I'm going to revert to that state.